docs/roomHandlers/fishingRoom.md

```diff
@@ -0,0 +1,41 @@
+# Fishing Room
+
+A room with a beach and a boat dock. The player can fish from dock or leave. Fishing may result in catching a fish.
+
+## States
+
+- **AtEntrance**: The player is on a dock with a fishing pole
+- **Fishing**: The player casts their fishing line in hopes to find a fish
+- **Caught**: The player now has a fish
+- **BrokenLine**: The player has a broken their fishing line and the fish got away
+- **FixLine**: The player can fix their broken line to fish again
+
+## Transitions
+
+- **leave**: The player leaves through the only available door
+- **fish**: The player casts their fishing line
+- **catch**: The player has caught a fish
+- **break**: The player lost the fish
+- **fix**: The player fixs their fishing pole so they can fish again
+
+## Diagram
+
+```mermaid
+stateDiagram-v2
+  [*] --> AtEntrance
+
+  AtEntrance --> Fishing: fish
+  AtEntrance --> [*]: leave
+
+  Fishing --> AtEntrance: default
+  Fishing --> Caught: catch
+  Fishing --> BrokenLine: break
+
+  Caught --> AtEntrance: default
+
+  BrokenLine --> AtEntrance: default
+  BrokenLine --> FixLine: fix
+  BrokenLine --> [*]: leave
+
+  FixLine --> Fishing: fish
+```
```

Comment:
I think I miscommunicated the `default` command. The `default` command is for inspecting the current state of the game. I think `Caught` is the only state where you would want `default` to say "you're back at the entrance" instead of repeating "you caught a fish!". The implied behavior of `default` is to loop back to the current state.

Here is my full suggestion for the `Fishing` and `fish` changes as well as an explicit `stop` command to get back to the entrance in some cases. I also removed FixLine since the command for fixing says the line is fixed.

```mermaid
stateDiagram-v2
  [*] --> AtEntrance

  AtEntrance --> Fishing: fish
  AtEntrance --> [*]: leave

  Fishing --> AtEntrance: stop
  Fishing --> Caught: fish (if catch)
  Fishing --> BrokenLine: fish (if break)
  Fishing --> Fishing: fish (else)

  Caught --> AtEntrance: default

  BrokenLine --> AtEntrance: stop
  BrokenLine --> AtEntrance: fix
  BrokenLine --> [*]: leave
```

Reply:
That makes more sense. I wasn't sure how `default` works and thought everything should be a state and have a default. I like this a lot more.